webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
+  private Long pollableTaskId;
+
+  @Override
+  public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    // Fetch the scheduled job and cast the properties
+    scheduledJob = scheduledJobRepository.findByJobKey(jobExecutionContext.getJobDetail().getKey());
+    ScheduledThirdPartySyncProperties scheduledJobProperties =
+        (ScheduledThirdPartySyncProperties) scheduledJob.getProperties();
+
+    logger.info(
+        "Third party sync for repository {} has started.", scheduledJob.getRepository().getName());
+
+    // Create ThirdPartySyncInput from scheduled job and properties
+    ThirdPartySyncJobInput thirdPartySyncJobInput =
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
+    } catch (Exception e) {
+      throw new JobExecutionException(e);
+    }
+  }
+
+  @Override
+  public void onSuccess(JobExecutionContext context) {
+    logger.info(
+        "Third-Party Sync succeeded for repository {}.", scheduledJob.getRepository().getName());
+
+    // Resolve PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              try {
+                pd.resolveIncident(scheduledJob.getId());
+              } catch (PagerDutyException e) {
```

Comment:
In relation to the above: it seems that here there's multiple errors that are being treated as part of the same category:
1. incident never existed in the first place (incorrect call from us to PD)
2. incident exists and is already resolved, nothing to resolve (incorrect call from us to PD)
3. incident exists and is not resolved -> PD error when attempting to resolve (and this is where we would actually want an error - and maybe a metric)

Solving the above problem re not doing unecessary/incorrect calls to PD would solve most of this.

Reply:
I agree with not needing to do these calls if an incident never occurred. 

However, PD doesn't throw any exceptions for attempting to resolve an incident from any of these points 1-3. The only exception here would be if the payload was incorrect or the PD servers are experiencing internal server errors. PD doesn't respond with errors if you try to resolve already resolved incidents or ones that don't exist to allow you to do things like this although we do really need a way to know if the job failed previously to ensure we aren't doing many external calls.

Reply:
The fact that PD doesn't throw exceptions doesn't detract from the fact that, for a job that's never failed for example, we'd still be sending hundreds/thousands of requests to PD over time to try to resolve an incident related to it (that never existed in the first place).

Similarly, we wouldn't want our system used for state tracking, once Mojito has delivered the core work of ingestion strings and delivering translations.

Let's track that work for a separate follow-on PR.
@mattwilshire - please feel free to resolve this once you've logged a ticket for this work (just to make sure we don't forget about this).

Reply:
Agreed @garionpin, I have logged a ticket for this.